### DISCUSSION TITLE
How to debug rust-based wasm under wasmedge?

### DISCUSSION THREAD

**Discussion by oliverzy** [WasmEdge/WasmEdge#3868] [General]
Do I need to build debug version of wasmedge?

**Comment by hydai** [comment]
If you want to dig into the execution of WasmEdge, then use the debug version. Otherwise, you can build debug enabled wasm and show the debug information.

**Comment by oliverzy** [comment]
Thanks, I am trying to debug wasmedge-quickjs in vscode with codelldb plugin.
Here is my lanuch config:
```
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "configurations": [
    {
      "type": "lldb",
      "request": "launch",
      "name": "Launch Wasmedge",
      "program": "wasmedge",
      "args": ["--dir", ".:.", "target/wasm32-wasi/debug/node.wasm"],
      "cwd": "${workspaceFolder}",
      "externalConsole": true,
      "stopOnEntry": true
    }
  ]
}
```

and I place a breakpoint in the entrypoint "src/main.rs", but it cannot break.

**Reply by hydai** [reply]
Currently, you cannot use WasmEdge as the debugging target. As a workaround, you may need to use debugging logs from the Rust side instead.
